src/software/ai/passing/cost_function.cpp

```diff
@@ -173,72 +142,64 @@ double calculateInterceptRisk(const Team& enemy_team, const Pass& pass,
-        (closest_point_on_pass_to_robot - pass.passerPoint()).length() / pass.speed());
-
-    // Check for division by 0
+    // Return early to avoid division by zero
     if (pass.speed() == 0)
     {
-        ball_time_to_closest_pass_point =
-            Duration::fromSeconds(std::numeric_limits<int>::max());
+        return 1.0;
     }
 
-    // Figure out how long the enemy robot and ball will take to reach the receive point
-    // for the pass.
-    Duration enemy_robot_time_to_pass_receive_position =
-        enemy_robot.getTimeToPosition(pass.receiverPoint());
-    Duration ball_time_to_pass_receive_position = pass.estimatePassDuration();
-
-    double robot_ball_time_diff_at_closest_pass_point =
-        ((enemy_robot_time_to_closest_pass_point + enemy_reaction_time) -
-         (ball_time_to_closest_pass_point))
-            .toSeconds();
-    double robot_ball_time_diff_at_pass_receive_point =
-        ((enemy_robot_time_to_pass_receive_position + enemy_reaction_time) -
-         (ball_time_to_pass_receive_position))
+    // We estimate the intercept by the risk that the enemy robot will get to the closest
+    // point on the pass before the ball
+    Point closest_interception_point = closestPoint(
+        enemy_robot.position(), Segment(pass.passerPoint(), pass.receiverPoint()));
+    Vector enemy_interception_vector =
+        closest_interception_point - enemy_robot.position();
+    // Take into account the enemy robot's radius for minimum min_interception_distance
+    // required to travel to intercept the pass.
+    double min_interception_distance =
+        std::max(0.0, enemy_interception_vector.length() - ROBOT_MAX_RADIUS_METERS);
+
+    const double ENEMY_ROBOT_INTERCEPTION_SPEED_METERS_PER_SECOND = 0.5;
+    double signed_1d_enemy_vel =
+        enemy_robot.velocity().dot(enemy_interception_vector.normalize());
+    double enemy_robot_time_to_interception_point_sec =
+        getTimeToTravelDistance(
+            min_interception_distance, ENEMY_ROBOT_MAX_SPEED_METERS_PER_SECOND,
+            ENEMY_ROBOT_MAX_ACCELERATION_METERS_PER_SECOND_SQUARED, signed_1d_enemy_vel,
+            ENEMY_ROBOT_INTERCEPTION_SPEED_METERS_PER_SECOND)
             .toSeconds();
+    // Scale the time to interception point by the enemy robot's interception capability
+    Duration enemy_robot_time_to_interception_point =
+        Duration::fromSeconds(enemy_robot_time_to_interception_point_sec *
+                              passing_config.enemy_interception_time_multiplier());
 
-    double min_time_diff = std::min(robot_ball_time_diff_at_closest_pass_point,
-                                    robot_ball_time_diff_at_pass_receive_point);
+    Duration ball_time_to_interception_point =
+        Duration::fromSeconds(distance(pass.passerPoint(), closest_interception_point) /
+                              pass.speed()) +
+        Duration::fromSeconds(passing_config.pass_delay_sec());
```

Comment:
this calculation seems duplicated below. consider refactoring this into a helper function `timeToPoint` that adds the `pass_delay_sec`

Reply:
I wasn't sure where it would make sense to include this function, but Ive now also added the TODO to both of them, since both of these lines and multiple other places in our code need to be updated to use a better ball model anyways, instead of a constant speed model.